runtime/vm/jfr.cpp

```diff
@@ -988,6 +992,35 @@ jfrClassLoadingStatistics(J9VMThread *currentThread)
 	}
 }
 
+void
+jfrThreadContextSwitchRate(J9VMThread *currentThread)
+{
+	PORT_ACCESS_FROM_VMC(currentThread);
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+
+	uint64_t switches;
```

Comment:
Please declare variables as the first line after the if statement.

Reply:
I don't see a problem here?

Reply:
Do you mean that the `switches` should be on top of port access?

Reply:
Yes.

Reply:
> Generally in vm code variable declarations go at the top to support old compilers. I see both patterns in this file so I'm not sure if that applies to cpp

This does not apply to C++ files. Due to legacy C rules (C89 and before), certain platforms will emit a warning (and error out because of `warnings-as-errors`) if C code has blocks wherein there are executable statements before variable declarations. From C99 and on, this was no longer a rule, but a convention for C code.

In C++, on the other hand, it is encouraged to declare variables close to the first use; there is no restriction/danger to declaring them anywhere in the block, provided they are declared before they are used of course.

Some VM C++ code looks like C code because it once was. But we don't need to restrict ourselves to legacy C standards going forward with C++.

All that being said, this is still technically valid C since the macros `PORT_ACCESS_FROM_VMC` and `OMRPORT_ACCESS_FROM_J9PORT` expand to variable declarations anyways.

However, we should be initializing all variables with an explicit value, which is missing from this line.